OpenEdX/Managers/PushNotificationsManager/Providers/FCMProvider.swift

```diff
@@ -6,13 +6,54 @@
 //
 
 import Foundation
+import Core
+import FirebaseCore
+import FirebaseMessaging
+import Swinject
 
-class FCMProvider: PushNotificationsProvider {
+class FCMProvider: NSObject, PushNotificationsProvider, MessagingDelegate {
+    
+    private var storage = Container.shared.resolve(CoreStorage.self)!
+    private let api = Container.shared.resolve(API.self)!
```

Comment:
I'm fine with this, but as we changed the dependency injection in the FCMListner, do we need to change it as well?

Reply:
Yes, it makes sense to provide them as dependencies on the init lever 👍